11-walkthrough-7.Rmd

```diff
@@ -1,209 +1,311 @@
+
 ```{r remove stop words}
 data(stop_words)
 
 tokens <- tokens %>%
-  anti_join(stop_words)
+  anti_join(stop_words, by = "word")
 ```
 
-# Counting words 
+Why does this work? Let's look closer. `inner_join` matches the observations in one dataset to another by a specified common variable. Any rows that don't have a match get dropped from the resulting dataset. `anti_join` does the same thing as `inner_join` except it drops matching rows and keeps the rows that *don't* match. This is convenient for our analysis because we want to remove rows from `tokens` that contain words in the `stop_words` dataset. When we call `anti_join`, we're left with rows that *don't* match words in the `stop_words` dataset. These remaining words are the ones we'll be analyzing.
+
+One final note before we start counting words: Remember when we first tokenized our dataset and we passed `unnest_tokens` the argument `output = word`? We conveniently chose `word` as our column name because it matches the column name `word` in the `stop_words` dataset. This makes our call to `anti_join` simpler because `anti_join` knows to look for the column named `word` in each dataset. 
 
-Though a basic step, assessing the most frequent terms can be powerful. The `topfeatures()` function can be helpful for this; simply pass `my_dfm` as the first argument to it to see the results.
+## Counting words 
+
+Now it's time to start exploring our newly cleaned dataset of tweets. Computing the frequency of each word and seeing which words showed up the most often is a good start. We can pipe `tokens` to the `count` function to do this: 
 
 ```{r count words}
 tokens %>% 
-    count(word, sort = TRUE)
+    count(word, sort = TRUE) 
+```
+
+We pass `count` the argument `sort = TRUE` to sort the `n` variable from the highest value to the lowest value. This makes it easy to see the most frequenly occuring words at the top. Not surprisingly, "tidytuesday" was the third most frequent word in this dataset. 
+
+We may want to explore further by showing the frequency of words as a percent of the whole dataset. We can do that by using `mutate` to add a column called `percent`. `percent` will divide `n` by `sum(n)`, which is the total number of words. Finally, will multiply the result by 100. 
```

Comment:
super minor - maybe mention something like "Calculating percentages (or proportions) like this can be useful for a range of data sources (such as for calculating what percentage of students are in each of a number of classes)."